optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/bi/BavetScoringBiConstraintStream.java

```diff
@@ -0,0 +1,166 @@
+import org.optaplanner.core.impl.score.inliner.ScoreInliner;
+import org.optaplanner.core.impl.score.inliner.UndoScoreImpacter;
+import org.optaplanner.core.impl.score.inliner.WeightedScoreImpacter;
+import org.optaplanner.core.impl.score.stream.bavet.BavetConstraint;
+import org.optaplanner.core.impl.score.stream.bavet.common.BavetNodeBuildPolicy;
+
+public final class BavetScoringBiConstraintStream<Solution_, A, B> extends BavetAbstractBiConstraintStream<Solution_, A, B> {
+
+    private final boolean positive;
+    private final boolean noMatchWeigher;
+    private final ToIntBiFunction<A, B> intMatchWeigher;
+    private final ToLongBiFunction<A, B> longMatchWeigher;
+    private final BiFunction<A, B, BigDecimal> bigDecimalMatchWeigher;
+
+    public BavetScoringBiConstraintStream(BavetConstraint<Solution_> bavetConstraint, boolean positive) {
+        this(bavetConstraint, positive, true, null, null, null);
+    }
+
+    public BavetScoringBiConstraintStream(BavetConstraint<Solution_> bavetConstraint, boolean positive, ToIntBiFunction<A, B> intMatchWeigher) {
+        this(bavetConstraint, positive, false, intMatchWeigher, null, null);
+        if (intMatchWeigher == null) {
+            throw new IllegalArgumentException("The matchWeigher (null) cannot be null.");
+        }
+    }
+
+    public BavetScoringBiConstraintStream(BavetConstraint<Solution_> bavetConstraint, boolean positive, ToLongBiFunction<A, B> longMatchWeigher) {
+        this(bavetConstraint, positive, false, null, longMatchWeigher, null);
+        if (longMatchWeigher == null) {
+            throw new IllegalArgumentException("The matchWeigher (null) cannot be null.");
+        }
+    }
+
+    public BavetScoringBiConstraintStream(BavetConstraint<Solution_> bavetConstraint, boolean positive, BiFunction<A, B, BigDecimal> bigDecimalMatchWeigher) {
+        this(bavetConstraint, positive, false, null, null, bigDecimalMatchWeigher);
+        if (bigDecimalMatchWeigher == null) {
+            throw new IllegalArgumentException("The matchWeigher (null) cannot be null.");
+        }
+    }
+
+    private BavetScoringBiConstraintStream(BavetConstraint<Solution_> bavetConstraint,
+            boolean positive, boolean noMatchWeigher,
+            ToIntBiFunction<A, B> intMatchWeigher, ToLongBiFunction<A, B> longMatchWeigher, BiFunction<A, B, BigDecimal> bigDecimalMatchWeigher) {
+        super(bavetConstraint);
+        this.positive = positive;
+        this.noMatchWeigher = noMatchWeigher;
+        this.intMatchWeigher = intMatchWeigher;
+        this.longMatchWeigher = longMatchWeigher;
+        this.bigDecimalMatchWeigher = bigDecimalMatchWeigher;
+    }
+
+    // ************************************************************************
+    // Node creation
+    // ************************************************************************
+
+    @Override
+    protected BavetScoringBiNode<A, B> createNode(BavetNodeBuildPolicy<Solution_> buildPolicy,
```

Comment:
The same comment as for the same method across other `BavetScoring*Node` classes